Windows: path separator causes 31 test failures (backslash vs forward slash)

## Environment
- OS: Windows 11
- Python: 3.14.3
- token-savior: latest main

## Summary
31 tests fail on Windows due to path separator mismatch.
All failures are one of two patterns:

**Pattern 1: Path separator**
AssertionError: assert 'src\core.py' == 'src/core.py'

**Pattern 2: Python import_graph is empty on Windows**
AssertionError: assert 0 > 0
where 0 = len({})
where {} = ProjectIndex(...).import_graph

## Impact
- Symbol navigation: ✅ works
- Persistent memory: ✅ works  
- Python import resolution: ❌ broken
- find_impacted_tests: ❌ broken

## Test results
31 failed, 1386 passed, 8 skipped
Affected test files:
- tests/test_impacted_tests.py
- tests/test_java_indexing.py
- tests/test_project_indexer.py (TestImportGraph)
- tests/test_symbol_embeddings.py
- tests/test_workflow_ops.py

--- a/src/token_savior/memory/symbol_embeddings.py
+++ b/src/token_savior/memory/symbol_embeddings.py
@@ -55,7 +55,7 @@ def _iter_symbols_in_file(py_path: Path, project_root: Path) -> Iterable[dict]:
         tree = ast.parse(source)
     except (OSError, SyntaxError):
         return
-    rel_path = str(py_path.relative_to(project_root))
+    rel_path = py_path.relative_to(project_root).as_posix()
 
     def _visit(node: ast.AST, prefix: str = "") -> Iterable[dict]:
         for child in ast.iter_child_nodes(node):

--- a/src/token_savior/project_indexer.py
+++ b/src/token_savior/project_indexer.py
@@ -285,7 +285,7 @@ class ProjectIndexer:
         total_classes = 0
 
         def _annotate_file(fpath: str) -> tuple[str, StructuralMetadata, float] | None:
-            rel_path = os.path.relpath(fpath, self.root_path)
+            rel_path = os.path.relpath(fpath, self.root_path).replace(os.sep, "/")
             try:
                 mtime = os.path.getmtime(fpath)
                 source = self._read_file(fpath)
@@ -393,7 +393,7 @@ class ProjectIndexer:
             if os.path.isabs(file_path)
             else os.path.join(self.root_path, file_path)
         )
-        rel_path = os.path.relpath(abs_path, self.root_path)
+        rel_path = os.path.relpath(abs_path, self.root_path).replace(os.sep, "/")
 
         idx = self._project_index
 
@@ -529,7 +529,7 @@ class ProjectIndexer:
             if os.path.isabs(file_path)
             else os.path.join(self.root_path, file_path)
         )
-        rel_path = os.path.relpath(abs_path, self.root_path)
+        rel_path = os.path.relpath(abs_path, self.root_path).replace(os.sep, "/")
 
         idx = self._project_index
         old_metadata = idx.files.get(rel_path)

